.github/workflows/ci.yml

```diff
@@ -17,23 +17,29 @@ on:
     tags:
       - '[0-9]+.[0-9]+.[0-9]+'
 
-env:
-  # Run only on PRs from the main repo, on scheduled, or triggered runs.
-  RUN_ON_PR_AND_SCHEDULE_AND_TRIGGER: github.event.pull_request.head.repo.full_name == github.repository || github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
-
 concurrency:
   # Only run one instance of this workflow in PRs, but allow multiple instances in develop.
   # https://stackoverflow.com/questions/74117321/if-condition-in-concurrency-in-gha
   group: ${{ github.workflow }}-${{ github.head_ref && github.ref || github.run_id }}
   cancel-in-progress: true
 
 jobs:
+
   ci-does-not-run-on-draft-pull-requests:
     runs-on: ubuntu-latest
     if: github.event.pull_request.draft == true
     steps:
       - run: echo "CI jobs won't run because this is a draft pull request."
 
+  ci-is-on-main-repo:
+    runs-on: ubuntu-latest
+    # job only runs on main repo where we have access to credentials, or as part of the release
+    if: | 
+      github.event.pull_request.head.repo.full_name == github.repository ||
```

Comment:
Can you walk me through this? 

1. Does `github.event.pull_request.head.repo.full_name == github.repository` mean our repo. Does it contain branch info (eg for `develop`)? I guess I'm concerned if it isn't for develop (eg the scheduled runs) it is too lenient.
2. Does `(github.event_name == 'push' && contains(github.ref, 'refs/tags/'))` imply a release since that's the only time we push a tag?

Reply:
Oh I see, the first part of the conditional is for a PR from our repo vs a PR from a fork?

Reply:
1. `github.event.pull_request.head.repo.full_name` is `Shinnnyshinshin/great_expectations` and `github.repository` is `great-expectations/great_expectations` so it tests whether we are in a fork or from the main repo
2. `refs/pull/8627/merge` is what we typically get in a PR, and yes as you said `refs/tags` implies a release